src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -761,11 +764,11 @@ public void delete(long participationId, boolean deleteBuildPlan, boolean delete
             var repositoryUri = programmingExerciseParticipation.getVcsRepositoryUri();
             String buildPlanId = programmingExerciseParticipation.getBuildPlanId();
 
-            if (deleteBuildPlan && buildPlanId != null) {
+            if ((deleteBuildPlan || profileService.isLocalVcsActive()) && buildPlanId != null) {
```

Comment:
this part is not strictly necessary, because deleteBuildPlan is empty in this case, but for consistency reasons and to stay compatible for potential future changes, we should keep it. Can you please adda a comment above?